constraint/pkg/core/constraints/constraints.go

```diff
@@ -11,13 +11,15 @@ import (
 // constraint is enforced. It is assumed that the author is comparing
 // two constraints with the same GVK/namespace/name.
 func SemanticEqual(c1 *unstructured.Unstructured, c2 *unstructured.Unstructured) bool {
-	s1, exists, err := unstructured.NestedMap(c1.Object, "spec")
-	if err != nil || !exists {
-		return false
+	if c1 == c2 {
+		return true
 	}
-	s2, exists, err := unstructured.NestedMap(c2.Object, "spec")
-	if err != nil || !exists {
+
+	if (c1 == nil) != (c2 == nil) {
 		return false
 	}
```

Comment:
nit: I don't think this second clause needs to be so complicated.

It could just be:

```golang
if c1 == nil || c2 == nil {
  return false
}
```

The first clause will catch when both pointers are `nil` (as they will be equal), and so we only need to worry about a single one of them being nil.  See this [playground](https://goplay.space/#MEENPSQFTVp) where I tested that.

Reply:
I found a more amusing solution that removes another if statement:
```
if c1 == nil || c2 == nil {
	return c1 == c2
}
```